langchain4j-core/src/main/java/dev/langchain4j/web/search/WebSearchOrganicResult.java

```diff
@@ -0,0 +1,204 @@
+ * This includes the title, link, content, and metadata of the web page.
+ * <p>
+ * These results are typically ranked by relevance to the search query.
+ * <p>
+ */
+public class WebSearchOrganicResult {
+
+    private final String title;
+    private final String link;
+    private final String content;
+    private final Map<String, String> resultMetadata;
+
+    /**
+     * Constructs a WebSearchOrganicResult object with the given content.
+     *
+     * @param content The content of the search result.
+     */
+    public WebSearchOrganicResult(String content) {
+        this(null, null, content, null);
+    }
+
+    /**
+     * Constructs a WebSearchOrganicResult object with the given content and link.
+     *
+     * @param content The content of the search result.
+     * @param link    The link associated with the search result.
+     */
+    public WebSearchOrganicResult(String content, String link) {
+        this(null, link, content, null);
+    }
+
+    /**
+     * Constructs a WebSearchOrganicResult object with the given title, link, and content.
+     *
+     * @param title   The title of the search result.
+     * @param link    The link associated with the search result.
+     * @param content The content of the search result.
+     */
+    public WebSearchOrganicResult(String title, String link, String content) {
+        this(title, link, content, null);
+    }
+
+    /**
+     * Constructs a WebSearchOrganicResult object with the given title, link, content, and result metadata.
+     *
+     * @param title           The title of the search result.
+     * @param link            The link associated with the search result.
+     * @param content         The content of the search result.
+     * @param resultMetadata  The metadata associated with the search result.
+     */
+    public WebSearchOrganicResult(String title, String link, String content, Map<String, String> resultMetadata) {
+        this.title = title;
+        this.link = link;
+        this.content = ensureNotBlank(content, "content");
+        this.resultMetadata = getOrDefault(resultMetadata, new HashMap<>());
+    }
+
+    /**
+     * Returns the title of the search result.
+     *
+     * @return The title of the search result.
+     */
+    public String title() {
+        return title;
+    }
+
+    /**
+     * Returns the link associated with the search result.
+     *
+     * @return The link associated with the search result.
+     */
+    public String link() {
+        return link;
+    }
+
+    /**
+     * Returns the content of the search result.
+     *
+     * @return The content of the search result.
+     */
+    public String content() {
+        return content;
+    }
+
+    /**
+     * Returns the result metadata associated with the search result.
+     *
+     * @return The result metadata associated with the search result.
+     */
+    public Map<String, String> resultMetadata() {
```

Comment:
```suggestion
    public Map<String, String> metadata() {
```
nitpick: "result" is redundant here, it is already in "result" class.
Also not consistent with the rest of the fields: content, not resultContent, etc